src/vm/functions/assets.rs

```diff
@@ -332,7 +353,70 @@ pub fn special_mint_asset(
     }
 }
 
-pub fn special_transfer_asset(
+/// The Stacks v205 version of mint_asset uses the actual stored size of the
+///  asset as input to the cost tabulation. Otherwise identical to v200.
+pub fn special_mint_asset_v205(
+    args: &[SymbolicExpression],
+    env: &mut Environment,
+    context: &LocalContext,
+) -> Result<Value> {
+    check_argument_count(3, args)?;
+
+    let asset_name = args[0].match_atom().ok_or(CheckErrors::BadTokenName)?;
+
+    let asset = eval(&args[1], env, context)?;
+    let to = eval(&args[2], env, context)?;
+
+    let nft_metadata = env
+        .contract_context
+        .meta_nft
+        .get(asset_name)
+        .ok_or(CheckErrors::NoSuchNFT(asset_name.to_string()))?;
+    let expected_asset_type = &nft_metadata.key_type;
+
+    let asset_size = byte_len_of_serialization(&asset.serialize());
+    runtime_cost(ClarityCostFunction::NftMint, env, asset_size)?;
+
+    if !expected_asset_type.admits(&asset) {
+        return Err(CheckErrors::TypeValueError(expected_asset_type.clone(), asset).into());
+    }
+
+    if let Value::Principal(ref to_principal) = to {
+        match env.global_context.database.get_nft_owner(
+            &env.contract_context.contract_identifier,
+            asset_name,
+            &asset,
+            expected_asset_type,
+        ) {
+            Err(Error::Runtime(RuntimeErrorType::NoSuchToken, _)) => Ok(()),
+            Ok(_owner) => return clarity_ecode!(MintAssetErrorCodes::ALREADY_EXIST),
+            Err(e) => Err(e),
+        }?;
+
+        env.add_memory(TypeSignature::PrincipalType.size() as u64)?;
+        env.add_memory(expected_asset_type.size() as u64)?;
```

Comment:
Argh, sorry, commented on the commit itself.  Just wanted some clarity (ha ha), this appears to be assessing the memory usage by the NFT's type signature, instead of the actual size (which is now used for the runtime cost). Is this intentional?

Reply:
This appears to apply to all of the new NFT methods.

Reply:
Yep -- good catch, we should use the actual size for the memory usage. Fixed in b83b8e1